src/nvim/os/env.c

```diff
@@ -223,6 +223,13 @@ void init_homedir(void)
 #endif
     homedir = xstrdup(var);
   }
+  else {
+    char IObuff_signed[MAXPATHL] = "";
+    if (uv_os_homedir(IObuff_signed, (size_t *)MAXPATHL)) {
```

Comment:
I would think that the use of `uv_os_homedir()` could replace most of what we're doing in this function, rather than being a fallback.  The only part that would still need special handling is setting `$HOME` if it isn't already present in the environment.